Source/ARTJsonLikeEncoder.m

```diff
@@ -427,6 +427,7 @@ - (ARTTokenRequest *)tokenRequestFromDictionary:(NSDictionary *)input error:(NSE
     [_logger verbose:@"RS:%p ARTJsonLikeEncoder<%@>: tokenRequestFromDictionary %@", _rest, [_delegate formatAsString], input];
 
     if (![input isKindOfClass:[NSDictionary class]]) {
+        *error = [ARTErrorInfo createWithCode:ARTStateRequestTokenFailed message:@"input provided has an invalid TokenRequest"];
```

Comment:
@tcard Do you agree with this? I add this check because an `authUrl` can return an empty object. The lib will try to use that empty object as a TokenRequest and then the callback isn't called because the `request` is nil.

I have some doubts mostly because the `authUrl` can return a `token` or a `tokenDetails` or even a `tokenRequest`. Maybe it's important to add a test checking a wrong use of `authUrl` on each case.